ClemBot.Bot/bot/utils/converters.py

```diff
@@ -54,18 +55,42 @@ async def convert(self, ctx: Context, duration: str) -> relativedelta:
 class Duration(DurationDelta):
     """Convert duration strings into UTC datetime.datetime objects."""
 
-    async def convert(self, ctx: Context, duration: t.Union[str, relativedelta]) -> datetime:
+    def __init__(self, ctx: Context, duration: t.Union[str, relativedelta]):
+        self.ctx = ctx
+        self.duration = duration
+        if isinstance(duration, relativedelta):
+            self.delta = duration
+        else:  # super.convert() is an async method - this is not
+            match = self.duration_parser.fullmatch(duration)
+            if not match:
+                raise ConversionError(f'`{duration}` is not a valid duration string.')
+            duration_dict = {unit: int(amount) for unit, amount in match.groupdict(default=0).items()}
+            self.delta = relativedelta(**duration_dict)
+
+    def as_future(self) -> datetime:
```

Comment:
this change will break the raffle cog, this file is for d.py command converters

Reply:
this change allows for "duration" in the future and in the past
the simple idea is that the cog takes in `DurationDelta` and then converts that to a `Duration` like this:
```py
async def some_cog(self, ctx: commands.Context, duration: DurationDelta):
    duration = Duration(ctx, duration)
    time: datetime = duration.as_future()  # or duration.as_past()
```
this was changed in `mute_cog.py` in this same PR and is used in the reminder cog.

it also provides consistent formatting for the delta, with the `__str__(self):` override using arrow's humanize func (which without, previously caused issues with mutes that were less than 1 minute).

i am more than happy to fix the raffle cog, though.

Reply:
The converter file is for dpy converters, what is a use case for  .as_past()?

Reply:
Originally these changes were in my purge cog, so you could give a duration of how far in the past the messages could be purged, since d.py's purge command used datetime, I thought it would be a good fit.

Reply:
That's actually not a bad reason, so then this should be two new dpy converters, one for future, one for past. The .as_future call for conversion in the method itself is unheeded and some weird boilerplate